feat(content): Expanding the description of Han Sizer jobs (#7805)
endless-sky/endless-sky: data/human/rim jobs.txt

	repeat
	description `In celebration of Han Sizer month, bring <fare> to all the marked systems and return to <destination> for <payment>.`
	passengers 2 10 .9
	substitutions

> 		"<on Longjump>" "somewhere nearby."

Reviewer: These substitutions are grammatically wrong! They won't integrate correctly into the sentence they're being used in.

Author: Really? They make:

- As they depart you hear them talking about celebrating their trip with a party somewhere nearby."
- As they depart you hear them talking about celebrating their trip with a party on Longjump."

Those both sound reasonable.

Reviewer: If they do resolve to that, that's good. I think they won't, though, and it should be double-checked in game. There might be an extra period at the end, for instance.

I'd this one really is good though, then good

Author: Fixed that. Thanks for pointing it out!

			has "flagship planet: Longjump"

> 	substitutions

Reviewer 2: It's `substitutions` plural because you can have multiple under the same node.
```suggestion
```

		distance 5 10
		attributes "rim"
		not attributes "station"
	on enter

> 		dialog `As you enter the system your passengers gather together to stare out the viewport. One of them pulls out a datapad and reads what sounds like a log entry by Han Sizer, describing the initial exploration of the system.`

Reviewer 2: Seems like it would be annoying for this `on enter` to pop up every time. I don't think this is the best way to give the player this info.

Author: That's why I only did it for *one* of these missions. So not every one of them is going to get this pop up, only some.

Reviewer 2: As I said in the review description, I'd rather just have a Remembrance Day-like mission conversation. Even if it's only one of the missions that has this on enter, it's still _every time_ you get this mission that it's going to appear.

Author: The `on enter` has been replaced with an on stopover, so its timing makes much better sense.

Reviewer 2: Still not what I asked for, though.

Author: But it is now on par with, say, the description of attending a research conference job, or the various retrieval jobs. It's not an onerous amount of text, and it *is* appropriate for what the job is. If it's a huge annoyance, than players can simply not do it - it's not something they are being forced to do. It's not even super profitable. It's an atmospheric/setting job that conveys a bit of culture of the region, and as such should be conveying a bit of what they're doing.

Having an additional mission as per your suggestion would be a great idea, and I'll be thinking about doing that. That is, however, a separate thing that I will aim to have in a standalone PR, and not a part of making these jobs a little more meaningful of what they are intended to convey.

Author: These are more complex jobs than the remembrance day things, and as such deserve the minute increase in screen time presented by a single `on stopover.` Also unlike the remembrance day jobs, the entire point of the job is the travel. They're not hiring the player to get from A to B, they're hiring the player to go through C. As such, it is appropriate that something actually happens while they are in transit.

As far as that goes, I don't really think the remembrance day jobs are a good model, because those are effectively meaningless. They have less character or impact than the "take a care package of X tons of Y stuff to a student on Z" jobs, and much less memorable. The Remembrance Day jobs are a step up from generic jobs, sure, but not by much. One explanatory conversation does *not* have any significant impact on justifying, explaining, or making meaningful all the varius remembrance day jobs I've done. They're all just bland "whatever" things that are basically indistinguishable from all the rest of the timed jobs. But that's a problem for another day, in a different PR.

Right now, I'm trying to take a small step in making *these* jobs slightly more relevant by including a tiny bit of flavor text, directions to where the player can find more, and a single `on stopover` in *one* of the jobs that illustrates that they're actually doing something on this trip. It's not just a "X passengers are taking up bunks while you wander around and happen to pass through a bunch of systems before they get off again."

	on complete
		payment 10000 170

> 		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect your payment of <payment>. As they depart you hear them talking about celebrating their trip with a party on Longjump.`

Reviewer 2: What if we are on Longjump, though? The other dialogs have this same issue. It seems like for these to not be awkward, Longjump would have to not be a valid source or stopover.

Author: Might be able to use a branch here to check for the player's location and alter the text if it is on longjump. Would that be acceptable?

Reviewer 2: Dialogs can't branch.

Author: Alright, sounds like a feature request, then.

Reviewer 3: ~~Use a substitution.~~

Author: substitution?

Author: Ah. https://github.com/endless-sky/endless-sky/wiki/CreatingSubstitutions

Author: I've got some substitutions now.

> 		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect your payment of <payment>. As they depart you hear them talking about celerbrating their trip with a party on Longjump.`

Reviewer 4: ```suggestion
		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect your payment of <payment>. As they depart you hear them talking about celebrating their trip with a party on Longjump.`
```

		not attributes "station"
	on complete
		payment 10000 170

> 		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect your payment of <payment>. As they depart you hear them talking about celebrating their trip with a party <on Longjump>.`

Reviewer 5: ```suggestion
		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect your payment of <payment>. As they depart, you hear them talking about celebrating their trip with a party <on Longjump>.`
```

	repeat
	description `In celebration of Han Sizer month, bring <fare> to all the marked systems and return to <destination> for <payment>.`
	passengers 2 10 .8
	substitutions

> 		"<on Longjump>" "near here."

Reviewer 6: ```suggestion
		"<on Longjump>" "near here"
```
Also, does that complete to `One is already uploading a travelogue to someone near here as they start walking away.` on line 145? It sound a bit weird, I think.

	repeat
	description `In celebration of Han Sizer month, bring <fare> to all the marked systems and return to <destination> for <payment>.`
	passengers 2 10 .8
	substitutions

> 		"<on Longjump>" "someone near here."

Reviewer: This will resolve, lower down there, to `One is already uploading a travelogue to someone someone near here. as they start walking away.`

Author: Thanks! Removed the extra "someone".

			has "flagship planet: Longjump"

> 	substitutions

Reviewer 2: ```suggestion
```

		not attributes "station"
	on complete
		payment 15000 170

> 		dialog `Your passengers are excited to have completed their journey around the Rim, and pay you the agreed fee of <payment>. One is already uploading a travelogue to someone <on Longjump> as they start walking away.`

Reviewer 4: ```suggestion
		dialog `Your passengers are excited to have completed their journey around the Rim and pay you the agreed fee of <payment>. One is already uploading a travelogue to someone <on Longjump> as they start walking away.`
```

		not attributes "station"
	on complete
		payment 20000 170

> 		dialog `Your guests thank you for taking them on a journey along the flightpath of Han Sizer, and cheerfully pay you <payment>. You notice one has a starchart open to look at Sabik's entry as they head off.`

Reviewer 4: ```suggestion
		dialog `Your guests thank you for taking them on a journey along the flightpath of Han Sizer, and cheerfully pay you <payment>. You notice one has a starchart open to Sabik's entry as they head off.`
```

		not attributes "station"
	on complete
		payment 20000 170

> 		dialog `Your guests thank you for taking them on a journey along the flightpath of Han Sizer and cheerfully pay you <payment>. You notice one has a starchart open to Sabik's entry as they head off.`

Reviewer 6: ```suggestion
		dialog `Your guests thank you for taking them on a journey along the flight path of Han Sizer and cheerfully pay you <payment>. You notice one has a star chart open to Sabik's entry as they head off.`
```

> 		dialog `Your guests thank you for taking them on a journey along the flightpath of Han Sizer, and cheerfully pay you <payment>. You notice one has a starchart open to Sabik's entry as they head off.`

Reviewer 4: ```suggestion
		dialog `Your guests thank you for taking them on a journey along the flightpath of Han Sizer and cheerfully pay you <payment>. You notice one has a starchart open to Sabik's entry as they head off.`
```
